### ISSUE TITLE
Clarification of use

### ISSUE THREAD

**Issue by neander-squirrel** [maxkueng/wyt#3]
I have similar use in my app as in the example below and I would expect "doing" appear on the logs every second:
 ```
var wyt = require("wyt")
const rateLimit = wyt(1, 1000);
 
async function doStuff(i) {
  await rateLimit();
  console.log('doing ' + i)
   
}

for (let i=0; i<5; i++){
  doStuff(i)
}
```
What I got instead is immediate execution 5 times.
I am trying to use wyt to rate limit my RPC node calls so it won't deny service, but it does not work as I've imagined.
How can I modify my test to get the expected result?

**Comment by maxkueng** [comment]
Hi. Your code is correct. There was a bug where the internal state was updated at the wrong time. I fixed it in 12d6239. I've published version 1.3.0 that contains the fix and updated the automated tests.

Additionally, `rateLimit()` now returns the time waited in milliseconds (#2) so you could do:
```js
async function doStuff(i) {
 const waited = await rateLimit();
 console.log(`doing ${i} and waited ${waited} milliseconds`)
}
```

Could you install the latest version and check if it now works the way you expect? If it doesn't, please reopen the issue.

**Comment by neander-squirrel** [comment]
Awesome, it works now! Thank you for the fast response!
